wizard

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env bash
+
+set -o allexport
+source "src/config"
+set +o allexport
+
+tmpLog="/tmp/swarmlet-install.log"
+instalLogLoc="/opt/swarmlet/install.log"
+
+### PKG Vars ###
+PKG_MANAGER="apt-get"
+PKG_CACHE="/var/lib/apt/lists/"
+UPDATE_PKG_CACHE="${PKG_MANAGER} update"
+PKG_INSTALL="${PKG_MANAGER} --yes --no-install-recommends install"
+PKG_COUNT="${PKG_MANAGER} -s -o Debug::NoLocking=true upgrade | grep -c ^Inst || true"
+SWARMLET_DEPS=(acl curl git apt-transport-https apache2-utils moreutils wget dialog)
+REPOSITORY="https://github.com/swarmlet/swarmlet.git"
+
+USERNAME="git"
+USER_ROOT="/home/$USERNAME"
+REPO_ROOT="/var/repo"
+DOCKER_SOCKET="/var/run/docker.sock"
+
+SWARMLET_ROOT="/opt/swarmlet"
+SWARMLET_CONFIG="$SWARMLET_ROOT/src/config"
+SWARMLET_FUNCTIONS="$SWARMLET_ROOT/src/functions"
+SWARMLET_HOOKS="$SWARMLET_ROOT/src/hooks"
+SWARMLET_MODULES_ROOT="$SWARMLET_ROOT/modules"
+
+
+AUTHORIZED_KEYS="/root/.ssh/authorized_keys" # TODO: make configurable
+SSH_COMMAND="command=\"$SWARMLET_ROOT/src/sshcommand $REPO_ROOT $SWARMLET_HOOKS\""
+SSH_OPTIONS="no-port-forwarding,no-X11-forwarding,no-agent-forwarding,no-pty,$SSH_COMMAND"
+
+SWARMLET_NETWORK_NAME="swarmlet-network"
+SWARMLET_NETWORK_DRIVER="overlay"
+
+# Find the rows and columns. Will default to 80x24 if it can not be detected.
+screen_size=$(stty size 2>/dev/null || echo 24 80)
+rows=$(echo ${screen_size} | awk '{print $1}')
+columns=$(echo ${screen_size} | awk '{print $2}')
+FPREFIX="=====>"
+PREFIX="----->"
+INDENT="      "
+
+declare -A DEFAULT_OPTS=(
+  [ROOT_DOMAIN]=$HOSTNAME   # (default $HOSTNAME) The domain to use for deployment of included services
+  [CF_EMAIL]=root@$HOSTNAME # The email which you are using for cloudflare registration
+  [INSTALLATION_MODE]=full  # (default full, options: full|minimal|bare) .. TODO
+  [INSTALL_BRANCH]=master   # (default master) The default branch to install
+  [CREATE_SWAP]=false       # (default false) Allocate 1GB of swap space
+  [INSTALL_ZSH]=false       # (default false) Install 'Oh My Zsh'
+  [NOOP]=                   # no-op for testing purposes
+)
+
+for KEY in "${!DEFAULT_OPTS[@]}"; do
+  [[ ! -z "${DEFAULT_OPTS[$KEY]}" ]] && export "$KEY=${DEFAULT_OPTS[$KEY]}"
+done
+# Divide by two so the dialogs take up half of the screen, which looks nice.
+r=$((rows / 2))
+c=$((columns / 2))
+# Unless the screen is tiny
+r=$((r < 20 ? 20 : r))
+c=$((c < 70 ? 70 : c))
+
+######## FIRST CHECK ########
+# Must be root to install
+echo "$FPREFIX Checking user"
+if [[ $EUID -eq 0 ]]; then
+  echo "$PREFIX You are root."
+else
+  echo "$PREFIX sudo will be used for the install."
+  # Check if it is actually installed
+  # If it isn't, exit because the install cannot complete
+  if [[ $(dpkg-query -s sudo) ]]; then
+    export SUDO="sudo"
+    export SUDOE="sudo -E"
```

Comment:
Why are we using `sudo -E` here? I thought I saw it somewhere before in your code, but now I cannot find it anymore.